.github/workflows/release.yml

```diff
@@ -0,0 +1,162 @@
+name: Release
+
+on:
+  pull_request:
+    branches:
+      - '*'  # must quote since "*" is a YAML reserved character; we want a string
+    paths-ignore:
+      - '.github/workflows/quarto-render.yml'
+      - '_quarto.yml'
+      - 'quarto-materials/*'
+      - '**/.md'
+      - 'tiledb/doxygen/source/*'
+      - 'tiledb/sm/c_api/tiledb_version.h'
+  push:
+    branches:
+      - dev
+      - 'release-*'
+    tags:
+      - '*'
+
+jobs:
+  Build-Release:
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: [macos-arm64, macos-x86_64, linux-x86_64, linux-x86_64-noavx2, windows-x86_64]
+        include:
+          - platform: windows-x86_64
+            os: windows-latest
+            triplet: x64-windows-release
+          - platform: linux-x86_64
+            os: ubuntu-latest
+            manylinux: true
+            triplet: x64-linux-release
+          - platform: linux-x86_64-noavx2
+            os: ubuntu-latest
+            cmake_args: -DCOMPILER_SUPPORTS_AVX2=OFF
+            triplet: x64-linux-release
+            manylinux: true
+          - platform: macos-x86_64
+            os: macos-latest
+            triplet: x64-osx-release
+          - platform: macos-arm64
+            os: macos-latest
+            cmake_args: -DCMAKE_OSX_ARCHITECTURES=arm64
+            triplet: arm64-osx-release
+    runs-on: ${{ matrix.os }}
+    container: ${{ matrix.manylinux && 'quay.io/pypa/manylinux2014_x86_64' || '' }}
+    env:
+      # We target macOS 11 at minimum according to CRAN policy.
+      # See https://stat.ethz.ch/pipermail/r-package-devel/2023q4/010078.html
+      MACOSX_DEPLOYMENT_TARGET: 11
```

Comment:
We need to use the same target in regular CI jobs.

Reply:
Moved to the CMake build options and removed setting the  `MACOSX_DEPLOYMENT_TARGET` environment variables.

Reply:
Reverted after moving back to targeting 10.14. Regular CI cannot run while targeting 10.14; some of the code will not be able to compile.